modules/structs/issue.go

```diff
@@ -190,6 +190,10 @@ func (l *IssueTemplateLabels) UnmarshalYAML(value *yaml.Node) error {
 	return fmt.Errorf("line %d: cannot unmarshal %s into IssueTemplateLabels", value.Line, value.ShortTag())
 }
 
+type IssueConfig struct {
+	BlankIssuesEnabled bool `json:"blank_issues_enabled" yaml:"blank_issues_enabled"`
+}
```

Comment:
What I've noticed from having to work with the GitHub templates already:
Collaborators who already have a lot of in-depth knowledge about a project tend to specify their issues with enough detail that they don't necessarily need to input all troubleshooting information.
Having to fill out the template can be pretty tiresome, especially for large projects like Gitea itself, and often leads to entering unnecessary information simply to satisfy the validation.
This means that later on, another value `for-contributors`, for example, could be added.
So, we could think about converting the type to (bool/)`string`, to support more use cases later on.
You don't need to implement the additional functionality I mentioned yet as that can be postponed.
However, I think converting it into a string can already be done in this PR.
As far as I know, it should not make a difference for the yaml parser.

Reply:
I agree that having such a setting would be useful, but @lunny requested compatibility with GitHub above, that's the reason it also searches in .github for the config. On GitHub, `blank_issues_enabled` is a boolean. That doesn't mean we can't add our own settings, but if we want compatibility with GitHub, we should not use a different type for the same setting as GitHub. Another setting would be the better solution here.

Reply:
But it would accept the same things as GitHub: `true, yes, on, off, no, false`,  (-> `strconv.ParseBool`)
and then more (i.e. `for-contributors`).
So it would still be compatible with GitHub?

Reply:
>  As far as I know, you don't need `"` around a YAML string…

That's true in most cases, but reserved keywords like `true` or `false` or numbers needs quotes to be accepted as string.

Reply:
Oh. Didn't know that.
But can our YAML Parser perhaps reconvert a boolean to a string?

Reply:
I don't know if the parser could do that. It looks like it needs to know the types of the values. But even when, it doesn't change the fact, that the result would not be compatible with GitHub, which don't accept something like `for-contributors`. It's way better and easier to just add another setting like `enable_blank_issues_for_contributors`.

Reply:
fwiw the parser seems to gracefully handle it, though I would need to test it in this context specifically.
https://go.dev/play/p/jNdWPWtK-Jz

Reply:
> It's way better and easier to just add another setting like enable_blank_issues_for_contributors.

There I strongly disagree.
That simply allows for an ungodly amount of edge cases and illegal states.

Reply:
> fwiw the parser seems to gracefully handle it, though I would need to test it in this context specifically. https://go.dev/play/p/jNdWPWtK-Jz

That's exactly what I hoped for.
This means we can still convert it without repercussions.
But I agree with @jolheiser, we should certainly ensure with a test that this behavior is fulfilled.

Reply:
I still think this is a bad idea. Even is that could parsed, this will confuse users and is still not comptaible with GitHub. I will keep the current behaviour for this PR. It is better to do is in another PR.

Reply:
We can just keep it only allow `true` and `false` in this PR. And in future PR, it can be allowed other values but we don't need to discuss it in this PR.